sync3/handler/connstate.go

```diff
@@ -473,6 +473,8 @@ func (s *ConnState) getInitialRoomData(ctx context.Context, roomSub sync3.RoomSu
 		var inviteState []json.RawMessage
 		// handle invites specially as we do not want to leak additional data beyond the invite_state and if
 		// we happen to have this room in the global cache we will do.
+		// Furthermore, rooms we have been invited to for the first time ever will not be in the global cache yet,
+		// which will cause errors below when we try calling functions on a nil metadata.
```

Comment:
Cross-linking: #85.

Off-topic here, but if you'll indulge me...

I can see that it's possible for the proxy to first learn of an room `R` is via an invite. But I don't understand what's special about invites here, as opposed to the other memberships. (Joins in particular being the most frequent, but also knocks. And what about kicks and bans?)

Reply:
Invites are special because they are the only membership state to occur prior to a join (excluding knocks which aren't implemented in any Element client, but have the same problems). Membership states which occur prior to a join are problematic for the proxy because there is no global state. The `invite_state` can change and get stale quickly, as it's only true at the point the invite was sent.

Join events will have global state associated with it. Leave/kick/bans all revert the state back to "not joined" which basically is just "remove this room from the list", i.e there is no room state to send to the client.